### PR TITLE
chore(repo): rename master branch to develop

### PR DESCRIPTION
**Pull request by akhilerm**
- As part of inclusive naming, work towards renaming branches
with name master to develop
- add ghcr to supported registries

### CODE REVIEW

**Review by RealHarshThakur — DISMISSED** [review]

**Comment by RealHarshThakur** [comment]
Just curious about this, we also have "slave" in the code base. Do we have to wait for Linux kernel to change the terminology(assuming they haven't) regarding that  so end users don't get confused about what we mean?

**Comment by akhilerm** [comment]
> Just curious about this, we also have "slave" in the code base. Do we have to wait for Linux kernel to change the terminology(assuming they haven't) regarding that so end users don't get confused about what we mean?

yep. That terminology is coming from sysfs in [kernel](https://github.com/torvalds/linux/blob/master/block/holder.c). so we will still have to use it when referring the devices.

**Comment by z0marlin** [comment]
@akhilerm need some clarity in the workflow. Are we moving the master branch to main or develop?

**Comment by akhilerm** [comment]
> @akhilerm need some clarity in the workflow. Are we moving the master branch to main or develop?

@z0marlin its develop. typo in the PR title. fixed it.

**Review by kmova — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ NDM uses the standard GitHub pull requests process to review and accept contribu
 * If you want to file an issue for a bug or feature request, please see [Filing a issue](#filing-an-issue)
 * If you have documentation improvement ideas, go ahead and create a pull request. See [Pull Request checklist](#pull-request-checklist)
 * If you would like to make code contributions, please start with [Setting up the Development Environment](#setting-up-your-development-environment).
-* If you would like to work on something more involved, please connect with the OpenEBS Contributors. See [OpenEBS Community](https://github.com/openebs/openebs/tree/master/community)
+* If you would like to work on something more involved, please connect with the OpenEBS Contributors. See [OpenEBS Community](https://github.com/openebs/openebs/tree/main/community)
```
Comment (z0marlin) [inline CONTRIBUTING.md:9]: should this link have `/tree/main/` in it? Shouldn't it be `/tree/develop/`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -15,10 +15,10 @@ NDM is an Apache 2.0 Licensed project and all your commits should be signed with
...
 * Fork the repository on GitHub.
-* Create a branch from where you want to base your work (usually master).
+* Create a branch from where you want to base your work (usually develop).
 * Make your changes. If you are working on code contributions, please see [Setting up the Development Environment](#setting-up-your-development-environment).
 * Relevant coding style guidelines are the [Go Code Review Comments](https://code.google.com/p/go-wiki/wiki/CodeReviewComments) and the _Formatting and style_ section of Peter Bourgon's [Go: Best Practices for Production Environments](http://peter.bourgon.org/go-in-production/#formatting-and-style).
-* Commit your changes by making sure the commit messages convey the need and notes about the commit. The commit message format followed for OpenEBS projects can be found [here](https://github.com/openebs/openebs/blob/master/contribute/git-commit-message.md).
+* Commit your changes by making sure the commit messages convey the need and notes about the commit. The commit message format followed for OpenEBS projects can be found [here](https://github.com/openebs/openebs/blob/main/contribute/git-commit-message.md).
```
Comment (z0marlin) [inline CONTRIBUTING.md:21]: `main` -> `develop` in the link?

**Review by z0marlin — COMMENTED** [review]
@akhilerm I see a lot of links referring branch `main`. Have marked two of them. Do we plan to have both `main` and `develop` branch? 

Also how do we plan to release these changes? Changes to the repository settings will have to be made afaik. We also need to create the branches right? Let's have a checklist for it? Apologies if there was a discussion for this already in the community sync-up since I couldn't attend it.

**Comment by kmova** [comment]
Putting a hold merge to check if renaming will break any raw url downloads that users might have.

File: CONTRIBUTING.md (hunk 1 above)
Comment (akhilerm) [reply CONTRIBUTING.md:9]: For repositories [openebs/openebs](https://github.com/openebs/openebs) and [linux-utils](https://github.com/openebs/linux-utils), `main` will be the default branch. for other repos `develop` will be used as release branches are created and releases are done from release branch. cc: @kmova

File: CONTRIBUTING.md (hunk 2 above)
Comment (akhilerm) [reply CONTRIBUTING.md:21]: same comment as above.

File: CONTRIBUTING.md (hunk 1 above)
Comment (akhilerm) [reply CONTRIBUTING.md:9]: Also, the branch creation/renaming will be done after the PR gets merged,

**Comment by kmova** [comment]
@akhilerm confirmed that renaming will not break any URLs and github will take care of redirecting as well updating the existing PRs to point to renamed branch. (Removing hold-merge)

**Review by z0marlin — APPROVED** [review]

**Pull request merged**